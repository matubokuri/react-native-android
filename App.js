/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */
import React, { Component } from 'react';
import { View } from 'react-native';

export default class FlexDirectionBasics extends Component<void, void> {
  render() {
    return (
      // Try setting `flexDirection` to `column`.
      <View style={{flex: 1, flexDirection: 'row'}}>
        <View style={{width: 50, height: 50, backgroundColor: 'powderblue'}} />
        <View style={{width: 50, height: 50, backgroundColor: 'skyblue'}} />
        <View style={{width: 50, height: 50, backgroundColor: 'steelblue'}} />
      </View>
    );
  }
}
