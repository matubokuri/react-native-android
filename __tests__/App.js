import 'react-native';
import React from 'react';
import { configure, shallow } from 'enzyme';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const dataSource = {
  "title": "The Basics - Networking",
  "description": "Your app fetched this from a remote endpoint!",
  "movies": [
    { "title": "Star Wars", "releaseYear": "1977" },
    { "title": "Back to the Future", "releaseYear": "1985" },
    { "title": "The Matrix", "releaseYear": "1999" },
    { "title": "Inception", "releaseYear": "2010" },
    { "title": "Interstellar", "releaseYear": "2014" }
  ]
}

fetch = jest.fn((url: string) => {
  return new Promise((resolve, reject) => {
    const value = {
      json: () => {
        return dataSource
      }
    }
    process.nextTick(() => resolve(value))
  })
})

it('renders correctly', async () => {
  expect.assertions(5)
  const tree = shallow(
    <App />
  );
  expect(tree.state('isLoading')).toBe(true)
  expect(tree.state('dataSource')).toBeUndefined()

  await tree.instance().componentDidMount()

  expect(tree.state('isLoading')).toBe(false)
  expect(tree.state('dataSource')).toHaveLength(dataSource.movies.length)

  tree.setState({ 'dataSource': dataSource.movies })

  expect(tree).toMatchSnapshot()
});
