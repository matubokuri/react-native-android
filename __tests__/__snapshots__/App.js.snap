// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "releaseYear": "1977",
          "title": "Star Wars",
        },
        Object {
          "releaseYear": "1985",
          "title": "Back to the Future",
        },
        Object {
          "releaseYear": "1999",
          "title": "The Matrix",
        },
        Object {
          "releaseYear": "2010",
          "title": "Inception",
        },
        Object {
          "releaseYear": "2014",
          "title": "Interstellar",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
